GVFS/GVFS.UnitTests/Common/LibGit2RepoPoolTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿using GVFS.Common;
+using GVFS.Common.Git;
+using GVFS.Tests.Should;
+using GVFS.UnitTests.Mock.Common;
+using NUnit.Framework;
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+
+namespace GVFS.UnitTests.Common
+{
+    [TestFixture]
+    public class LibGit2RepoPoolTests
+    {
+        [TestCase]
+        public void PoolConstructsThenDisposesRepos()
+        {
+            MockTracer tracer = new MockTracer();
+            int size = 3;
+            TimeSpan dueTime = TimeSpan.FromMilliseconds(1);
+            TimeSpan period = TimeSpan.FromMilliseconds(1);
+
+            BlockingCollection<object> threadTriggers = new BlockingCollection<object>();
+            BlockingCollection<object> disposalTriggers = new BlockingCollection<object>();
+
+            using (LibGit2RepoPool pool = new LibGit2RepoPool(tracer, () => new MockLibGit2Repo(disposalTriggers), size, dueTime, period))
+            {
+                for (int i = 0; i < size; i++)
+                {
+                    new Thread(() => pool.TryInvoke(repo => threadTriggers.TryTake(out object _, 5000), out bool result)).Start();

```

Comment:
I was having trouble with the unit test here, but figured out how to ensure that we create and dispose of _exactly_ `size` repos. This loop starts the threads, and then we only allow one to finish at a time by adding items to the `threadTriggers` collection.